pkg/ebpf/events_pipeline.go

```diff
@@ -237,14 +239,109 @@ func (t *Tracee) decodeEvents(outerCtx context.Context, sourceChan chan []byte)
 	return out, errc
 }
 
-// shouldProcessEvent decides whether or not to drop an event before further processing it
-func (t *Tracee) shouldProcessEvent(event trace.Event) bool {
+// computeScopes iterates through the scopes that do the filtering in user space, checking whether an event should be considered.
+// If it should, it sets the respective offset to 1, otherwise it sets it to 0.
+// Finally it returns the bitmask of computed scopes.
+func (t *Tracee) computeScopes(event *trace.Event) uint64 {
 	eventID := events.ID(event.EventID)
+	origMatchedScopes := event.MatchedScopes
+	matchedScopes := event.MatchedScopes
+
+	for filterScope := range t.config.FilterScopes.UserSpaceMap() {
+		bitOffset := uint(filterScope.ID)
+
+		// Events submitted with matching scopes.
+		// The scope must have its bit cleared when it does not match.
+		if origMatchedScopes != 0 && utils.HasBit(origMatchedScopes, bitOffset) {
+			if filterScope.CheckEventsInUserSpace &&
+				!filterScope.IsEventTraceable(eventID) {
```

Comment:
In a future PR, checking if an event is traceable for some scope can be done using a BPF map